shared/tests/utility/math/TestCoordinates.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2013 NUbots <nubots@nubots.net>
+ */
+
+#include <Eigen/Core>
+#include <array>
+#include <catch.hpp>
+#include <cfloat>
+#include <cmath>
+#include <fmt/format.h>
+#include <iomanip>
+#include <utility>
+#include <yaml-cpp/yaml.h>
+
+#include "utility/math/coordinates.hpp"
+#include "utility/support/yaml_expression.hpp"
+
+using utility::math::coordinates::cartesianToSpherical;
+using utility::math::coordinates::sphericalToCartesian;
+using utility::support::Expression;
+using utility::support::resolve_expression;
+
+
+/**
+ * @author Liam Craft
+ */
+
+static constexpr double ERROR_THRESHOLD = 1e-6;
+
+/** Test Data */
+static const YAML::Node test_values = YAML::LoadFile("tests/CoordinateTests.yaml");
+
+// cartesian to spherical
+SCENARIO("Cartesian coordinates can be converted to spherical coordinates", "[utility][math][coordinates]") {
+    GIVEN("A set of cartesian coordinates and a set of expected values") {
+        // cart input
+        static const std::array<Eigen::Vector3d, 200> cart_coords =
+            resolve_expression<Eigen::Vector3d, 200>(test_values["cartesianInput"]);
+        // cart to spher results
+        static const std::array<Eigen::Vector3d, 200> cart_to_spher_expected =
+            resolve_expression<Eigen::Vector3d, 200>(test_values["cartToSpher_results"]);
+        for (size_t i = 0; i < cart_coords.size(); i++) {
+            WHEN("Cartesian coordinates are converted to spherical coordinates") {
+                static const Eigen::Vector3d result = cartesianToSpherical(cart_coords.at(i));
+                THEN("The results of the calculation should be approximately equal to the expected values") {
+                    // NOTE:We use approx because floating points dont like being compared
+                    // at the epsilon value of the expected results
```

Comment:
I don't quite understand this comment. What do you mean by "at the epsilon value of the expected results"? I piece together you're referring to the return value of the `Approx(<thing>).epsilon(ERROR_THRESHOLD)`, but calling it the `epsilon value of the expected results` confuses me. Naming these variables `epsilon_<letter>` confuses me too, since catch's name for the `epsilon` function itself seems like a misnomer (on catch's behalf) to me.

Reply:
Yeah i don't understand it either. Looks like i deleted a line in the middle or something.